lms/djangoapps/django_comment_client/forum/tests.py

```diff
@@ -531,9 +531,133 @@ def test_group_info_in_ajax_response(self, mock_request):
         )
 
 
+@patch('requests.request')
+class SingleThreadContentGroupTestCase(ContentGroupTestCase):
+    def assert_can_access(self, user, discussion_id, thread_id, should_have_access):
+        """
+        Verify that a user has access to a thread within a given
+        discussion_id when should_have_access is True, otherwise
+        verify that the user does not have access to that thread.
+        """
+        request = RequestFactory().get("dummy_url")
+        request.user = user
+        mako_middleware_process_request(request)
+
+        def call_single_thread():
+            return views.single_thread(
+                request,
+                unicode(self.course.id),
+                discussion_id,
+                thread_id
+            )
+
+        if should_have_access:
+            self.assertEqual(call_single_thread().status_code, 200)
+        else:
+            with self.assertRaises(Http404):
+                call_single_thread()
+
+    def assert_searched_with_discussion_ids(self, mock_request, expected_commentable_ids):
+        """
+        Verify that the comments service was searched for threads with
+        the expected discussion ids (passed to the comments service as
+        'commentable_ids').
+        """
+        mock_request.assert_called_with(
+            'get',
+            StringEndsWithMatcher('threads'),
+            headers=ANY,
+            timeout=ANY,
+            data=None,
+            params=PartialDictMatcher({
+                'course_id': unicode(self.course.id),
+                'commentable_ids': ','.join(self.course.top_level_discussion_topic_ids + expected_commentable_ids)
+            })
+        )
+
+    def test_staff_user(self, mock_request):
+        """
+        Verify that the staff user can access threads in the alpha,
+        beta, and global discussion modules.
+        """
+        def assert_searched_correct_modules():
+            self.assert_searched_with_discussion_ids(
+                mock_request,
+                [self.beta_module.discussion_id, self.global_module.discussion_id, self.alpha_module.discussion_id]
+            )
+
+        thread_id = "test_thread_id"
+        mock_request.side_effect = make_mock_request_impl("dummy content", thread_id)
+
+        for discussion_module in [self.alpha_module, self.beta_module, self.global_module]:
+            self.assert_can_access(self.staff_user, discussion_module.discussion_id, thread_id, True)
+            assert_searched_correct_modules()
```

Comment:
Good question. While we technically only need to check that we searched the correct modules one time, it's nice to verify that this behavior is consistent and does not depend on which thread the user is querying.